dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -226,4 +264,180 @@ private void checkModelPluralForm(String apiCategory, String model) {
+			@RequestParam MultiValueMap<String, Object> parameters) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+			
+		Link link = linkTo(this.getClass(), apiCategory, model).slash("search").slash(searchMethod).withSelfRel();
+		DSpaceRestRepository repository = utils.getResourceRepository(apiCategory, model);
+		Page<DSpaceResource<T>> resources = null;
+		boolean searchEnabled = false;
+		boolean searchMethodFound = false;
+		for (Method method : repository.getClass().getMethods()) {
+			SearchRestMethod ann = method.getAnnotation(SearchRestMethod.class);
+			if (ann != null) {
+				searchEnabled = true;
+				String name = ann.name();
+				if (name.isEmpty()) {
+					name = method.getName();
+				}
+				if (StringUtils.equals(name, searchMethod)) {
+					searchMethodFound = true;
+					resources = ((Page<T>) executeQueryMethod(repository, parameters, method, pageable, sort, assembler)).map(repository::wrapResource);
+					break;
+				}
+			}
+		}
+		if (!searchMethodFound && searchEnabled) {
+			throw new RepositorySearchMethodNotFoundException(model, searchMethod);
+		}
+		if (!searchEnabled) {
+			throw new RepositorySearchNotFoundException(model);
+		}
+		PagedResources<DSpaceResource<T>> result = assembler.toResource(resources, link);
+		return result;
+	}
+	
+	private boolean haveSearchMethods(DSpaceRestRepository repository) {
+		for (Method method : repository.getClass().getMethods()) {
+			SearchRestMethod ann = method.getAnnotation(SearchRestMethod.class);
+			if (ann != null) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/*
+	 * Adapted from org.springframework.data.rest.webmvc.RepositorySearchController.executeQueryMethod(RepositoryInvoker, MultiValueMap<String, Object>, Method, DefaultedPageable, Sort, PersistentEntityResourceAssembler)
+	 */
+	private Object executeQueryMethod(DSpaceRestRepository repository,
+			MultiValueMap<String, Object> parameters, Method method, Pageable pageable, Sort sort,
+			PagedResourcesAssembler assembler) {
+
+		MultiValueMap<String, Object> result = new LinkedMultiValueMap<String, Object>(parameters);
+		MethodParameters methodParameters = new MethodParameters(method, new AnnotationAttribute(Param.class));
+
+		for (Entry<String, List<Object>> entry : parameters.entrySet()) {
+
+			MethodParameter parameter = methodParameters.getParameter(entry.getKey());
+
+			if (parameter == null) {
+				continue;
+			}
+
+			result.put(parameter.getParameterName(), entry.getValue());
+		}
+
+		return invokeQueryMethod(repository, method, result, pageable, sort);
+	}
+	
+	/*
+	 * Adapted from org.springframework.data.repository.support.ReflectionRepositoryInvoker.invokeQueryMethod(Method, MultiValueMap<String, ? extends Object>, Pageable, Sort)
+	 */
+	public Object invokeQueryMethod(DSpaceRestRepository repository, Method method, MultiValueMap<String, ? extends Object> parameters, Pageable pageable,
```

Comment:
I would also move these out of the `RestResourceController` class into a reflection / invoker utility class.